fff

```diff
@@ -171,6 +175,37 @@ read_dir() {
     cur_list=("${list[@]}")
 }
 
+# Convert number to a human readable format (with K, M, G suffixes)
+human_readable()
+{
+    local s=$1
+    if [[ $s -lt 1024 ]]; then
+        echo "$s\\e[33m"
+    elif [[ $s -lt 1048576 ]]; then
+        echo "$((s / 1024))\\e[32mK"
+    elif [[ $s -lt $((1048576 * 1024)) ]]; then
+        echo "$((s / 1048576))\\e[32mM"
+    elif [[ $s -lt $((1048576 * 1048576)) ]]; then
+        echo "$((s / (1048576 * 1024)))\\e[32mG"
+    else
+        echo "$s"
+    fi
+}
+
+# Show file size in a human readable format, only for a regular files
+human_readable_size()
+{
+    local f="$1"
+    if [[ -L "$f" ]]; then
+        echo ""
+    elif [[ -f "$f" ]]; then
+        echo "\\e[33m$(human_readable "$(stat --printf=%s "$f")")"
```

Comment:
iirc `stat` has no POSIX specification. Do these flags work everywhere?

Reply:
No, this is a coreutils stat command. Will fix it to POSIX compliant

Reply:
Here's what I've found (we may have to use different flags per OS):

- GNU: https://linux.die.net/man/1/stat
- FreeBSD: https://www.freebsd.org/cgi/man.cgi?query=stat&sektion=1
- OpenBSD: https://man.openbsd.org/stat.1
- macOS: https://ss64.com/osx/stat.html
- Busybox:

```
stat
stat [OPTIONS] FILE...
Display file (default) or filesystem status

        -c fmt  Use the specified format
        -f      Display filesystem status
        -L      Follow links
        -t      Display info in terse form

Valid format sequences for files:

 %a     Access rights in octal
 %A     Access rights in human readable form
 %b     Number of blocks allocated (see %B)
 %B     The size in bytes of each block reported by %b
 %d     Device number in decimal
 %D     Device number in hex
 %f     Raw mode in hex
 %F     File type
 %g     Group ID of owner
 %G     Group name of owner
 %h     Number of hard links
 %i     Inode number
 %n     File name
 %N     File name, with -> TARGET if symlink
 %o     I/O block size
 %s     Total size, in bytes
 %t     Major device type in hex
 %T     Minor device type in hex
 %u     User ID of owner
 %U     User name of owner
 %x     Time of last access
 %X     Time of last access as seconds since Epoch
 %y     Time of last modification
 %Y     Time of last modification as seconds since Epoch
 %z     Time of last change
 %Z     Time of last change as seconds since Epoch

Valid format sequences for file systems:

 %a     Free blocks available to non-superuser
 %b     Total data blocks in file system
 %c     Total file nodes in file system
 %d     Free file nodes in file system
 %f     Free blocks in file system
 %i     File System ID in hex
 %l     Maximum length of filenames
 %n     File name
 %s     Block size (for faster transfer)
 %S     Fundamental block size (for block counts)
 %t     Type in hex
 %T     Type in human readable form


Read more at: https://www.commandlinux.com/man-page/man1/busybox.1.html#lbAG
```

Reply:
Another option is using `ls` (*which has a POSIX specification*) to print the information.